python_scripts/linear_models_regularization.py

```diff
@@ -266,12 +266,25 @@
 # of `alpha` will decrease the weight values. A negative value of `alpha` would
 # actually enhance large weights and promote overfitting.
 #
+# ```{note}
+# Here, we only focus on numerical features. For categorical features, it is
+# generally common to omit scaling when features are encoded with a
+# `OneHotEncoder` since the feature values are already on a similar scale.
+#
+# However, there is no good practice since scaling has an interaction with the
+# regularization as well. For instance, scaling categorical features that are
+# imbalanced (e.g. more occurrences of a specific category) allows to apply a
+# similar penalty to each of the category. However, scaling such features in
+# presence of rare category could be problematic (i.e. division by a very
+# small std. dev.) and it will introduce some numerical issue.
```

Comment:
```suggestion
# imbalanced (e.g. more occurrences of a specific category) would even out
# the impact of regularization to each category. However, scaling such features
# in the presence of rare categories could be problematic (i.e. division by a very
# small standard deviation) and it can therefore introduce numerical issues.
```